server/src/main/java/com/thoughtworks/go/server/domain/user/WhitelistFilter.java

```diff
@@ -41,8 +44,18 @@ public String name() {
     }
 
     @Override
-    public boolean isPipelineVisible(CaseInsensitiveString pipeline) {
-        return null != pipelines && !pipelines.isEmpty() && pipelines.contains(pipeline);
+    public boolean isPipelineVisible(CaseInsensitiveString pipeline, StageInstanceModel latestStage) {
```

Comment:
@ibnc I think another valid option would be filtering on the client side, and it might be simpler?